src/app/components/play-round/play-round.component.ts

```diff
@@ -16,17 +16,20 @@ import { DomSanitizer } from '@angular/platform-browser';
 })
 export class PlayRoundComponent implements OnInit {
   flowerArMarkers$: Observable<RoundMarker[]> = combineLatest([
+    this.studentRoundService.currentBeeSpecies$,
     this.studentRoundService.currentFlowers$,
     this.studentRoundService.currentBeePollen$,
     this.studentRoundService.recentFlowerInteractions$,
   ]).pipe(
-    map(([flowers, beePollen, recentInteractions]) =>
+    filter(([bee]) => !!bee),
+    map(([bee, flowers, beePollen, recentInteractions]) =>
       flowers.map((flower, index) =>
         roundMarkerFromRoundFlower(
           flower,
           index + 1,
           beePollen,
           recentInteractions,
+          !bee.flowers_accepted.map(acceptedFlower => acceptedFlower.id).includes(flower.species.id)
```

Comment:
I think you should pass the bee object into this function and have the function do this part.

Reply:
I am not averse to either way. Is there a benefit in passing primitive type rather than object reference ? I feel like marker.ts doesn't need to know the BeeSpecies type, so I didn't want to import it for a single parameter.

Reply:
We are already passing in the entire flower, it feels like if we're calling a function to do the work we shouldn't be doing any of the logic here.

Reply:
Alright, moved the logic to the roundMarkerFromRoundFlower function